pkg/router/controller/host_admitter.go

```diff
@@ -183,10 +178,14 @@ func (p *HostAdmitter) addRoute(route *routeapi.Route) error {
 		p.claimedWildcards.RemoveRoute(wildcardKey, displacedRoute)
 
 		msg := fmt.Sprintf("a route in another namespace holds host %s", displacedRoute.Spec.Host)
-		p.recorder.RecordRouteRejection(displacedRoute, "SubdomainAlreadyClaimed", msg)
+		p.recordRejection(displacedRoute, "HostAlreadyClaimed", msg)
 		p.plugin.HandleRoute(watch.Deleted, displacedRoute)
 	}
 
+	if len(route.Spec.WildcardPolicy) == 0 {
```

Comment:
Some of the tests fail if this is removed - we need to explicitly set it otherwise the default is empty string in the tests. AFAICT, the default code only kicks when it does a conversion to/from the store. The unit tests don't really do that - they just instantiate a new object and use it.